x/ibc/07-tendermint/types/evidence.go

```diff
@@ -73,6 +74,12 @@ func (ev Evidence) GetHeight() int64 {
 	return int64(math.Min(float64(ev.Header1.Height), float64(ev.Header2.Height)))
 }
 
+// GetTime returns the timestamp at which misbehaviour occurred
+func (ev Evidence) GetTime() time.Time {
+	minTime := int64(math.Min(float64(ev.Header1.Time.UnixNano()), float64(ev.Header2.Time.UnixNano())))
```

Comment:
could you add comment in the godoc on what is occurring here?

Reply:
Is there any TM doc we can link here for why the misbehaviour is marked as occurring at the creation of the first header and not the creation of the second header? 

My logic would be:
- first vote is valid
- second vote is time at misbehaviour

Reply:
> Is there any TM doc we can link here for why the misbehaviour is marked as occurring at the creation of the first header and not the creation of the second header?

AFAIK you can submit the headers in any order and get the same result. Otherwise, we should rename the fields to `validHeader` and `misbehaviourHeader`

Reply:
I think I just mean using `math.Max` instead of `math.Min`. I'm not sure which is desirable, but my worry is using `math.Min` could open attacks for producing an invalid header outside of the evidence age range

Reply:
> I think I just mean using `math.Max` instead of `math.Min`. I'm not sure which is desirable, but my worry is using `math.Min` could open attacks for producing an invalid header outside of the evidence age range

Yeah, I totally agree that this should be documented and specified. The reason why we use Min on the height is that we want to freeze the client at the lowest possible height if there's misbehaviour. That said, under the same logic it would make sense to use `Max` for time to prevent invalid headers from not being valid if they are outside the max-age, especially since we don't have a `FrozenTimestamp` field for clients.

Thoughts @AdityaSripal @cwgoes?

Reply:
> My logic would be:
   first vote is valid
   second vote is time at misbehaviour

AFAIK, nothing constrains an equivocating validator from signing a second header with an earlier time, so we can't assume this.

> AFAIK you can submit the headers in any order and get the same result. Otherwise, we should rename the fields to validHeader and misbehaviourHeader

Yes, and for our purposes there should be no "preferred" header, it's misbehaviour either way.

> The reason why we use Min on the height is that we want to freeze the client at the lowest possible height if there's misbehaviour.

I don't understand this reasoning, what does the timestamp have to do with the height?

I believe it makes sense to use `Max` here so that the evidence is considered valid if either header is within the timerange, but we should double-check with Tendermint.